src/deserialize.zig

```diff
@@ -120,19 +120,22 @@ pub fn deserialize(comptime T: type, serialized: []const u8, out: *T) !usize {
                     return 1 + size + size_size;
                 }
             } else {
-                if (serialized[0] < rlpByteListShortHeader) {
-                    return try deserialize(ptr.child, serialized[0..1], &out.*[0]);
-                } else if (serialized[0] < rlpByteListLongHeader) {
-                    const size = @as(usize, serialized[0] - rlpByteListShortHeader);
+                if (serialized[0] < rlpListShortHeader) {
+                    return error.NotAnRLPList;
+                }
+
+                if (serialized[0] < rlpListLongHeader) {
+                    const size = @as(usize, serialized[0] - rlpListShortHeader);
```

Comment:
I'm not entirely sure why the current (previous) code was that way -- maybe a copy-paste from other case?

Reply:
or I thought of a different type of list, but nothing comes to mind right now so it's probably an error indeed.